app/controllers/sponsor/contributions_controller.rb

```diff
@@ -1,47 +1,47 @@
 class Sponsor::ContributionsController < ApplicationController
-  # before_action :find_contributions, only: [:index, :show, :create]
+  before_action :find_contributions, only: [:show, :create, :edit, :update]
 
   def index
     @contributions = Contribution.all
-    # authorize @contributions
     @contributions = policy_scope [:sponsor, Contribution]
     # authorize [:sponsor, @contribution]
   end
 
   def show
-    # @contribution made available at the bottom
-    @contribution = Contribution.find(params[:id])
+    # @contribution made available with private method
   end
 
-  def new
-    @contribution = Contribution.new
-    # authorize @contribution
-  end
+  # As of now sponsor should not be able to create contributions
+  # def new
+  #   @contribution = Contribution.new
+  #   # authorize @contribution
+  # end
 
-  def create
-    @contribution = Contribution.new(contribution_params)
-    @contribution.user = contribution_user
-    # authorize @contribution
-    if @contribution.save
-      redirect_to contribution_path(@contribution)
-    else
-      render :new
-    end
-  end
+  # def create
+  #   @contribution = Contribution.new(contribution_params)
+  #   @contribution.user = contribution_user
+  #   # authorize @contribution
+  #   if @contribution.save
+  #     redirect_to contribution_path(@contribution)
+  #   else
+  #     render :new
+  #   end
+  # end
 
   def edit
-     # @contribution made available at the bottom
+    # @contribution made available with private method
+    # sponsor should be able to change :status enum
   end
 
   def update
-     # @contribution made available at the bottom
+     # @contribution made available with private method
      @contribution.update(contribution_params)
      # redirect to show page
      redirect_to contribution_path
   end
 
   def destroy
```

Comment:
we don't need the sponsor to be able to delete contributions, they will only assign a status to the contribution (accept or reject)

so you can comment out or delete this method